testbed/tests/widgets/test_button.py

```diff
@@ -46,46 +45,33 @@ async def test_press(widget, probe):
 
     # Set up a mock handler, and press the button again.
     handler = Mock()
-    # TODO: can't use assert_set_get, because getattr returns the wrapped handler, which
-    # is an implementation detail that we shouldn't expose.
-    # https://github.com/beeware/toga/pull/804 may be relevant.
-    setattr(widget, "on_press", handler)
+    widget.on_press = handler
     probe.press()
+    await probe.redraw()
     handler.assert_called_once_with(widget)
 
 
-@mark.skipif(
-    current_platform in {"windows"},
-    reason="color reset on transparent not implemented",
-)
 async def test_background_color_transparent(widget, probe):
     "Buttons treat background transparency as a color reset."
     widget.style.background_color = TRANSPARENT
+    await probe.redraw()
     assert_color(probe.background_color, None)
 
 
-@mark.skipif(
-    current_platform in {"android", "iOS"},
-    reason="await redraw() not implemented",
-)
-@mark.skipif(
-    current_platform in {"linux"},
-    reason="resizes not applying correctly",
-)
 async def test_button_size(widget, probe):
     "Check that the button resizes"
     # Container is initially a non-flex row box.
-    # Initial button size is small, based on content size.
+    # Initial button size is small (but non-zero), based on content size.
     await probe.redraw()
-    assert 50 <= probe.width <= 100
-    assert probe.height <= 50
+    assert 10 <= probe.width <= 150, f"Width ({probe.width}) not in range (10, 150)"
+    assert 10 <= probe.height <= 50, f"Height ({probe.height}) not in range (10, 50)"
```

Comment:
As discussed, these messages should be generated automatically by pytest's "assertion rewriting", but that isn't currently working on Android. But since this PR is blocking all the other widget testing, let's fix that separately.